CommandLine/protos/server.proto

```diff
@@ -86,3 +87,8 @@ message SystemInfo {
   optional string author = 4; ///< Name of the system author or maintainer (e.g, "John Doe")
   optional string target = 5; ///< The target-platform of the system. Only used by compiler descriptions presently.
 }
+
+message OutputReply {
+  optional string out = 1; ///< All available output read from the standard output stream.
+  optional string err = 2; ///< All available output read from the standard error stream.
```

Comment:
Another thing LGM/ENIGMA do right now is transmit a progress update. So, consider...
```proto
message ProgressMessage {
  optional float progress = 1;  // In range 1-100 (you can use 0-1, but take a stance, here)
  optional string message = 2;  // The message to display inside of or after the progress bar.
}

message OutputReply {
  repeated LogMessage message = 1;  // Compilation messages logged since the last update.
  optional ProgressMessage progress = 2;  // Current compilation progress.
}
```

Reply:
Ok: 36dac03f55f24fc0d46082042cfd1de594e6d234

But I have many questions about how I am actually supposed to get those to RGM through emake. Right now we actually don't load any of the callbacks ENIGMA uses to send stuff to LGM. 

https://github.com/enigma-dev/enigma-dev/blob/27a85e52a9f0e30e19990d51e5b59a0a9363cac3/CompilerSource/backend/EnigmaCallbacks.h#L23
https://github.com/enigma-dev/enigma-dev/blob/27a85e52a9f0e30e19990d51e5b59a0a9363cac3/CompilerSource/backend/JavaCallbacks.h#L54

First of all, I'm not sure which of those two is the correct one to use here if any. Regardless, we can also see that the emake plugin class is just throwing away callbacks right now.

https://github.com/enigma-dev/enigma-dev/blob/27a85e52a9f0e30e19990d51e5b59a0a9363cac3/CommandLine/emake/EnigmaPlugin.cpp#L88

That said, I am not sure how to approach this. Should I make the plugin class implement the callbacks and then store that state? This would include a queue of log messages synchronized with a semaphore. Then you would have to initialize one plugin instance per game compile thread. Does that sound right?

Reply:
Ok: 56479aa539e435826763ffbde9f8d08a4420b360 and ebf156b57f7226c03941c7430678c67090f0fbac

I gave it a shot, and it worked. Now I just need a critique of that approach and help with a few issues, mainly the fact that because I am only taking log messages from AppendFrame IDE callback, I am getting excess newlines in the output.